src/axom/inlet/examples/document_generation.cpp

```diff
@@ -0,0 +1,146 @@
+
+  currField = inlet->addInt("thermal_solver/mesh/parallel", "parallel value");
+  currField->required(true);
+
+  currField = inlet->addInt("thermal_solver/order", "thermal solver order");
+  currField->required(true);
+  
+  currField = inlet->addString("thermal_solver/timestepper", "thermal solver timestepper");
+  currField->required(true);
+
+  currField = inlet->addString("thermal_solver/u0/type", "description for u0 type");
+  currField->required(true);
+
+  currField = inlet->addString("thermal_solver/u0/func", "description for u0 func"); 
+  currField->required(true);
+  
+  currField = inlet->addString("thermal_solver/kappa/type", "description for kappa type");
+  currField->required(true);
+
+  currField = inlet->addDouble("thermal_solver/kappa/constant", "description for kappa constant");
+  currField->required(true);
+
+  // Add description to solver table by using the addTable function
+  auto table = inlet->addTable("thermal_solver/solver", "This is the solver sub-table in the thermal_solver table");
+
+  // Can also add fields through a table
+
+  currField = table->addDouble("rel_tol", "description for solver rel tol");
+  currField->required(true);
+  
+  currField = table->addDouble("abs_tol", "description for solver abs tol");
+  currField->required(true);  
+
+  currField = table->addInt("print_level", "description for solver print level");
+  currField->required(true);  
+
+  currField = table->addInt("max_iter", "description for solver max iter");
+  currField->required(true);  
+  
+  currField = table->addDouble("dt", "description for solver dt");
+  currField->required(true); 
+
+  currField = table->addInt("steps", "description for solver steps");
+  currField->required(true);
+
+  // Checking contents of inlet
+
+  findStr("thermal_solver/mesh/filename", inlet);
+  findStr("thermal_solver/timestepper", inlet);
+  findStr("thermal_solver/u0/type", inlet);
+  findStr("thermal_solver/u0/func", inlet);
+  findStr("thermal_solver/kappa/type", inlet);
+
+  findInt("thermal_solver/mesh/serial", inlet);
+  findInt("thermal_solver/mesh/parallel", inlet);
+  findInt("thermal_solver/order", inlet);
+  findInt("thermal_solver/solver/print_level", inlet);
+  findInt("thermal_solver/solver/max_iter", inlet);
+  findInt("thermal_solver/solver/steps", inlet);
+
+  findDouble("thermal_solver/solver/dt", inlet);
+  findDouble("thermal_solver/solver/abs_tol", inlet);
+  findDouble("thermal_solver/solver/rel_tol", inlet);
+  findDouble("thermal_solver/kappa/constant", inlet);
+  
+  // Generating documentation
+  inlet->writeDocs();
+}
+
+int main(int argc, char** argv) {
+  CLI::App app {"Description here"};
+  bool docsEnabled{false};
+  app.add_flag("--enableDocs", docsEnabled, "Enables documentation generation");
+
+  std::string strOption;
+  auto opt = app.add_option("--deck", strOption, "Path to input deck file");
+  opt->check(CLI::ExistingFile);
+
+  CLI11_PARSE(app, argc, argv);
+
+  DataStore ds;

```

Comment:
You could move this into exampleInlet

Reply:
Just the datastore declaration or the CLI11_PARSE as well?

Reply:
Just the datastore